### ISSUE TITLE
Collisions between block params and html tags 

### ISSUE THREAD

**Issue by mistahenry** [emberjs/ember.js#17219]
As of 3.3
```javascript
images: []
```
```handlebars
{{#each images as |img|}}
    <img
      class="d-block w-100"
      src={{img}}
      alt="Post Image"
    >
 {{/each}}
```
will fail  because  `<img>` attempts to invoke the `|img|` as a component. The same code works in 
 < 3.2 

The error in 3.3 is:
> You cannot use 'https://avatars0.githubusercontent.com/u/4096934?s=460&v=4' as a component name. Component names must contain a hyphen.

3.4 in up
>Assertion Failed: fullName must be a proper full name

Please see [this twiddle](https://ember-twiddle.com/40809ac0e78bd57c89681901c5bbd598?openFiles=templates.application.hbs%2C)

That such a scenario errors is understandable, but there could be some improvement in the error messaging. Alternatively, I could see a new rule in [ember template lint](https://github.com/ember-template-lint/ember-template-lint) that errors when block params collide with html tags as sufficient. If the template rule is the choice, I could open an issue there and would be happy to take a stab at it.

**Comment by rwjblue** [comment]
I commented over in https://github.com/emberjs/ember.js/issues/16826#issuecomment-408504435 with the details, but I'll duplicate here (for ease of future travelers):

> Is this considered a breaking change?

Yes, it is a breaking change and we are working on the fix.

> I'm also a little confused by the RFC as to whether this is allowed (and supposed to work), or allowed at build-time (but breaks at run-time).

The current behavior matches what the RFC stated (essentially that a block param named `button` + `<button></button>` in the blocks template, would assume that the block param was a yielded closure component and attempt to render it as such), but based on feedback (here and in other issues) we determined that the result is definitely something we consider a breaking change.

The current plan (after discussing with the core team on 2018-07-20) is:

* Ensure that all block params that "shadow" normal HTML elements inside their blocks, continue to render the HTML element (and do not assume they should "invoke" a yielded component).
* Issue a deprecation when this scenario is detected (with `until: '4.0.0'`)
* Ensure the linter is setup for newly generated Ember apps that will lint against using confusing block param names (where the block param _could be_ an HTML element name, and that HTML element is invoked in the blocks template).
* Backport these fixes to ~~Ember 3.3~~ Ember 3.4 LTS and the stable release at the time of fixing.

---

> I could see a new rule in ember template lint that errors when block params collide with html tags as sufficient.

FWIW, this rule exists ([no-shadowed-elements](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-shadowed-elements.md)) and is enabled by default in the recommended rule set in a `ember new` generated app.
